libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -691,7 +699,18 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		for (size_t i = 0; i < arguments.size(); ++i)
 			args += convert(*arguments[i], *parameterTypes[i]).stackSlots();
 
-		if (functionDef)
+		if (_functionCall.annotation().kind == FunctionCallKind::StructConstructorCall)
+		{
+			TypeType const& type = dynamic_cast<TypeType const&>(*_functionCall.expression().annotation().type);
+			auto const& structType = dynamic_cast<StructType const&>(*type.actualType());
+
+			define(_functionCall) <<
+				m_utils.allocateAndInitializeMemoryStructFunction(structType) <<
+				"(" <<
+				joinHumanReadable(args) <<
```

Comment:
Are the types already properly converted?

Reply:
See line 700